api/auth/client.go

```diff
@@ -18,171 +18,157 @@ type Auth struct {
 	api restapi.Connector
 }
 
-// New creates a new auth client instance, using the
-// argument SDK API client.
+// New auth client constructor.
 func New(api restapi.Connector) *Auth {
 	return &Auth{api: api}
 }
 
-// AuthStatus get microservice status
-func (store *Auth) AuthStatus() (*common.ServiceStatus, error) {
-	status := &common.ServiceStatus{}
+// MARK: Status
+// Status get auth microservice status.
+func (c *Auth) Status() (*response.ServiceStatus, error) {
+	status := &response.ServiceStatus{}
 
-	_, err := store.api.
+	_, err := c.api.
 		URL("/auth/api/v1/status").
 		Get(status)
 
 	return status, err
 }
 
+// MARK: Identity Provider
 // CreateIdpClient creates a new identity provider client configuration.
-func (store *Auth) CreateIdpClient(idpClient *IDPClient) (IDstruct, error) {
-	idpClientId := IDstruct{}
+func (c *Auth) CreateIdpClient(idpClient *IdpClient) (*response.Id, error) {
+	idpClientResponse := &response.Id{}
 
-	_, err := store.api.
+	_, err := c.api.
 		URL("/auth/api/v1/idp/clients").
-		Post(&idpClient, &idpClientId)
+		Post(&idpClient, &idpClientResponse)
 
-	return idpClientId, err
+	return idpClientResponse, err
 }
 
 // UpdateIdpClient updates existing identity provider client configuration definition.
-func (store *Auth) UpdateIdpClient(idpClient *IDPClient, idpID string) error {
-
-	_, err := store.api.
-		URL("/auth/api/v1/idp/clients/%s", idpID).
+func (c *Auth) UpdateIdpClient(idpClient *IdpClient, idpId string) error {
+	_, err := c.api.
+		URL("/auth/api/v1/idp/clients/%s", url.PathEscape(idpId)).
 		Put(&idpClient)
 
 	return err
 }
 
-// IdpClient fetches existing identity provider client configuration.
-func (store *Auth) IdpClient(idpID string) (*IDPClient, error) {
-	idpClient := &IDPClient{}
+// GetIdpClient get existing identity provider client configuration.
+func (c *Auth) GetIdpClient(idpId string) (*IdpClient, error) {
+	idpClient := &IdpClient{}
 
-	_, err := store.api.
-		URL("/auth/api/v1/idp/clients/%s", idpID).
+	_, err := c.api.
+		URL("/auth/api/v1/idp/clients/%s", url.PathEscape(idpId)).
```

Comment:
Since escaping is an URL operation, can it be done inside `func (client *tClient) URL(templatePath string, args ...interface{}) CURL` so that we do not need to repeat this?